### PR TITLE
doc: added a note on pulling task image from private repo

### PR DESCRIPTION
**Pull request by adelmoradian**
# Changes

Added a note on pulling task image from private repo.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by jerop** [comment]
@adelmoradian please sign the CLA: https://github.com/tektoncd/pipeline/pull/6426#issuecomment-1481142928

File: docs/tasks.md [hunk 1]
```diff
@@ -175,6 +175,9 @@ The following requirements apply to each container image referenced in a `steps`
   though Tekton runs `Steps` sequentially.
   For more detail, see [Compute Resources in Tekton](./compute-resources.md).
 
+**Note:** If the image referenced in the `step` field is from a private registry, `TaskRuns` or `PipelineRuns` that consume the task
+          must provide the `imagePullSecrets` in a [podTemplate](./podtemplates.md).
+
```
Comment (jerop) [inline docs/tasks.md:180]:
```suggestion
**Note:** If the image referenced in the `step` field is from a private registry, `TaskRuns` or `PipelineRuns` that use the task must provide the `imagePullSecrets` in a [podTemplate](./podtemplates.md).
```

**Review by jerop — COMMENTED** [review]
Thanks for the contribution @adelmoradian, and welcome to the Tekton community!

/meow

Please add more details in the commit message, for example:

```
doc: added a note on pulling task image from private repo

if a task uses an image from a private repo, then `imagePullSecrets`
must be provided at runtime - added note to the tasks docs 
```

Standards: https://github.com/tektoncd/community/blob/main/standards.md#commit-messages

**Comment by adelmoradian** [comment]
@jerop amended the commit message and signed the CLA!

**Review by afrittoli — COMMENTED** [review]
/lgtm

**Pull request merged**